File: Factory/Python/Principal.py
import ShapeFactory

if __name__ == '__main__':
    mi_shape = ShapeFactory.shapeFactory()

    rectangulo = mi_shape.get_Shape('r', 4)
    rectangulo.getLados()
    rectangulo.draw()
    cuadrado = mi_shape.get_Shape('s', 2)   
    cuadrado.getLados()
    cuadrado.draw()

File: Factory/Java/Principal.java
public class Principal {

    public static void main(String[] args){
        ShapeFactory shapeFactory = new ShapeFactory();

        // Construyendo un objeto de tipo triangulo
        Shape triangulo = shapeFactory.getShape("TRIANGLE");
        triangulo.draw(); // Dibujando el circulo

        System.out.println("");
        // Construyendo un objeto tipo rectangulo
        Shape rectangulo = shapeFactory.getShape("RECTANGLE");
        rectangulo.draw(); // dibujando el rectangulo

        System.out.println("");
        // Construyendo un objeto tipo cuadrado
        Shape cuadrado = shapeFactory.getShape("SQUARE");
        cuadrado.draw(); // dibujando el cuadrado


    }
    
}
